lib/blockchain/bloom.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Blockchain.Bloom do
+  use Bitwise
+
+  @spec create(binary()) :: integer()
+  def create(data) when is_binary(data) do
+    bloom(0, data)
+  end
+
+  @spec add(integer(), binary()) :: integer()
+  def add(bloom_number, data) when is_binary(data) do
+    bloom(bloom_number, data)
+  end
+
+  @spec contains?(integer(), binary()) :: boolean()
+  def contains?(current_bloom, val)
+      when is_integer(current_bloom) and
+           is_binary(val) do
+    bloom = create(val)
+
+    (bloom &&& current_bloom) == bloom
+  end
+
+  @spec bloom(integer(), binary()) :: integer()
+  defp bloom(number, data) do
+    bits =
+      data
+      |> sha3_hash
+      |> bit_numbers
+
+    number |> add_bits(bits)
+  end
+
+  @spec sha3_hash(binary()) :: binary()
+  defp sha3_hash(data) do
+    data |> :keccakf1600.sha3_256
+  end
+
+  @spec add_bits(integer(), [integer()]) :: integer()
+  defp add_bits(bloom_number, bits) do
+    bits
+    |> Enum.reduce(bloom_number, fn(bit_number, bloom) ->
+      bloom ||| (1 <<< bit_number)
```

Comment:
for example, let's bit_number = 151. We have to set 151 th bit of the 2048 bit bloom filter
```elixir
iex(1)> (1 <<< 151) |> Integer.to_string(2)
"10..0" // 150 zeros
```
||| - bitwise OR operator